packages/central-server/src/kobo/startSyncWithKoBo.js

```diff
@@ -52,50 +52,48 @@ const writeKoboDataToTupaia = async (transactingModels, koboData, syncGroupCode)
 
   let newSyncTime = dataServiceSyncGroup.sync_cursor;
   let numberOfSurveyResponsesCreated = 0;
-  for (const koboSyncResponse of koboData) {
-    for (const [surveyCode, responses] of Object.entries(koboSyncResponse)) {
-      const survey = await transactingModels.survey.findOne({ code: surveyCode });
-
-      for (const responseData of responses) {
-        if (responseData.eventDate > newSyncTime) {
-          newSyncTime = responseData.eventDate;
-        }
-        const entity = await transactingModels.entity.findOne({ code: responseData.orgUnit });
-        if (!entity) {
-          await dataServiceSyncGroup.log(
-            `Skipping KoBo sync for record id ${responseData.event}: unknown entity ${responseData.orgUnit}`,
-          );
-          continue;
-        }
-
-        const surveyResponse = await transactingModels.surveyResponse.create({
-          id: generateId(),
-          survey_id: survey.id,
-          user_id: apiUser.user_account_id,
-          assessor_name: responseData.assessor || 'KoBo Integration',
-          entity_id: entity.id,
-          start_time: responseData.eventDate,
-          end_time: responseData.eventDate,
-          data_time: responseData.eventDate,
-        });
-
-        const questions = await transactingModels.question.find({
-          code: Object.keys(responseData.dataValues),
-        });
-        const questionByCode = keyBy(questions, 'code');
-
-        await transactingModels.answer.createMany(
-          Object.entries(responseData.dataValues).map(([questionCode, answer]) => ({
-            id: generateId(),
-            type: questionByCode[questionCode].type,
-            survey_response_id: surveyResponse.id,
-            question_id: questionByCode[questionCode].id,
-            text: answer,
-          })),
-        );
+  for (const [surveyCode, responses] of Object.entries(koboData)) {
+    const survey = await transactingModels.survey.findOne({ code: surveyCode });
```

Comment:
Mostly just whitespace changes here, took out the top level of the loop

Reply:
Ah yikes what an ugly bug, good spotting and solution. 
I haven't looked too much into it but is this a type issue? The fact that that typescript didn't complain when the getter disappeared?